### PR TITLE
Implement assert_liveliness()

### PR DESCRIPTION
**Pull request by mm318**
Implement assert_liveliness().

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: rmw/include/rmw/rmw.h [hunk 1]
```diff
@@ -251,32 +256,6 @@ rmw_publisher_count_matched_subscriptions(
   const rmw_publisher_t * publisher,
   size_t * subscription_count);
 
-/// Retrieve the actual qos settings of the publisher.
```
Comment (mm318) [inline rmw/include/rmw/rmw.h:254]: This is from https://github.com/ros2/rmw/commit/16d3cf4aaf3e964459b816aa32b0ff69bb95a477. We don't need to include this right now.

File: rmw/include/rmw/rmw.h [hunk 2]
```diff
@@ -182,6 +182,11 @@ RMW_WARN_UNUSED
 rmw_ret_t
 rmw_destroy_node(rmw_node_t * node);
 
+RMW_PUBLIC
+RMW_WARN_UNUSED
+rmw_ret_t
+rmw_node_assert_liveliness(const rmw_node_t * node);
```
Comment (nburek) [inline rmw/include/rmw/rmw.h:207]: Add the documentation for these new functions

File: rmw/src/event.c [hunk 3]
```diff
@@ -24,14 +24,23 @@ extern "C" {
 rmw_event_t
 rmw_get_zero_initialized_event(void)
 {
-  const rmw_event_t event = {
-    .implementation_identifier = NULL,
-    .data = NULL,
-    .event_type = RMW_EVENT_INVALID
+  return (const rmw_event_t) {
```
Comment (nburek) [inline rmw/src/event.c]: I think Devin just had to fix this to correct the formatting on this PR: https://github.com/ros2/rmw/pull/171/files#diff-87bfd7387c3d42809cf922e0ebbb347fR27

Let's make sure we're not reverting that.

**Review by nburek — CHANGES_REQUESTED** [review]
Looks good. Just get the function documentation added and I'll ship it.

File: rmw/include/rmw/rmw.h (hunk 2 above)
Comment (mm318) [reply rmw/include/rmw/rmw.h:207]: Done.

File: rmw/src/event.c (hunk 3 above)
Comment (mm318) [reply rmw/src/event.c]: Done.

File: rmw/include/rmw/rmw.h [hunk 4]
```diff
@@ -182,6 +182,31 @@ RMW_WARN_UNUSED
 rmw_ret_t
 rmw_destroy_node(rmw_node_t * node);
 
+/// Manually assert that this node is alive (for RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE)
+/**
+ * If the rmw Liveliness policy is set to RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE, the creator of 
+ * this node may manually call `assert_liveliness` at some point in time to signal to the rest
+ * of the system that this Node is still alive.
+ * This function must be called at least as often as the qos_profile's liveliness_lease_duration
```
Comment (nburek) [inline rmw/include/rmw/rmw.h]:
```suggestion
 * 
```

File: rmw/include/rmw/rmw.h [hunk 5]
```diff
@@ -296,6 +295,31 @@ rmw_ret_t
 rmw_publish_serialized_message(
   const rmw_publisher_t * publisher, const rmw_serialized_message_t * serialized_message);
 
+/// Manually assert that this Publisher is alive (for RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC)
+/**
+ * If the rmw Liveliness policy is set to RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC, the creator of
+ * this publisher may manually call `assert_liveliness` at some point in time to signal to the rest
+ * of the system that this Node is still alive.
+ * This function must be called at least as often as the qos_profile's liveliness_lease_duration
```
Comment (nburek) [inline rmw/include/rmw/rmw.h]:
```suggestion
 *
```

**Review by nburek — APPROVED** [review]
Approved pending the last two nitpick comments about "must be called" not being true.

File: rmw/include/rmw/rmw.h (hunk 4 above)
Comment (mm318) [reply rmw/include/rmw/rmw.h]: Done.

File: rmw/include/rmw/rmw.h (hunk 5 above)
Comment (mm318) [reply rmw/include/rmw/rmw.h]: Done.

**Pull request merged**